tests/tier2/test_reduce_labels_to_label_edges.cpp

```diff
@@ -0,0 +1,51 @@
+#include "cle.hpp"
+
+#include <array>
+#include <gtest/gtest.h>
+
+class TestReduceLabelsToLabelEdges : public ::testing::TestWithParam<std::string>
+{
+protected:
+  std::array<uint32_t, 8 * 8 * 1> output;
+
+  std::array<float, 8 * 8 * 1> valid = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 2, 0, 1, 0,
+                                         1, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 3, 3, 3, 4, 4, 4, 0, 0, 3, 0, 3,
+                                         4, 0, 4, 0, 0, 3, 3, 3, 4, 4, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+  std::array<float, 8 * 8 * 1> input = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 2, 0, 1, 1,
+                                         1, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 3, 3, 3, 4, 4, 4, 0, 0, 3, 3, 3,
+                                         4, 4, 4, 0, 0, 3, 3, 3, 4, 4, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+};
+
+TEST_P(TestReduceLabelsToLabelEdges, execute)
+{
+  std::string param = GetParam();
+  cle::BackendManager::getInstance().setBackend(param);
+  auto device = cle::BackendManager::getInstance().getBackend().getDevice("", "all");
+  device->setWaitToFinish(true);
+
+  auto gpu_input = cle::Array::create(5, 4, 2, 3, cle::dType::FLOAT, cle::mType::BUFFER, device);
+  gpu_input->write(input.data());
```

Comment:
The `gpu_input` `Array::create()` does not correspond to the array CPU you are trying to process

`output` is of type `uint8` of shape `8*8*1`
`valid` and `input` are of type `float`
and you create `gpu_input` as an array of `(5,4,2)` of type `float`

Reply:
I let you fix the test :)